src/primitives/relocatable.ts

```diff
@@ -110,3 +110,25 @@ export class Relocatable {
     return this.offset;
   }
 }
+
+/**
+ * Subclass of Relocatable, specific to the Allocation Pointer (Ap) and the Frame Pointer (Fp)
+ * These CairoVM registers are considered relocatables must can only have segment index equal to 1
+ * as they always point to the execution segment.
+ */
+export class VmPointer extends Relocatable {
```

Comment:
Should we rename this the ExecutionSegment? (Cairo paper page 41)

Reply:
but it's a relocatable no? the execution segment is the segment that has index 1?

Reply:
Mmmmm true, I just find the naming a little strange

Reply:
Hmm true, 
`FpOrAp`?
MemoryPointer?
Registers?

Reply:
MemoryPointer or FpOrAp sounds good, preferences for MemoryPointer personally.

Reply:
ok